audformat/core/utils.py

```diff
@@ -1317,84 +1317,14 @@ def union(
 ) -> pd.Index:
     r"""Create union of index objects.
 
-    Index objects must be conform to
-    :ref:`table specifications <data-tables:Tables>`.
-    Otherwise use
-    :func:`audformat.utils.union_misc`.
-
-    If at least one object is segmented, the output is a segmented index.
-
-    Args:
-        objs: index objects conform to
-            :ref:`table specifications <data-tables:Tables>`
-
-    Returns:
-        union of index objects
-
-    Raises:
-        ValueError: if one or more objects are not conform to
-            :ref:`table specifications <data-tables:Tables>`
-
-    Example:
-        >>> union(
-        ...     [
-        ...         filewise_index(['f1', 'f2', 'f3']),
-        ...         filewise_index(['f2', 'f3', 'f4']),
-        ...     ]
-        ... )
-        Index(['f1', 'f2', 'f3', 'f4'], dtype='string', name='file')
-        >>> union(
-        ...     [
-        ...         segmented_index(['f2'], [0], [1]),
-        ...         segmented_index(['f1', 'f2'], [0, 1], [1, 2]),
-        ...     ]
-        ... )
-        MultiIndex([('f2', '0 days 00:00:00', '0 days 00:00:01'),
-                    ('f1', '0 days 00:00:00', '0 days 00:00:01'),
-                    ('f2', '0 days 00:00:01', '0 days 00:00:02')],
-                   names=['file', 'start', 'end'])
-        >>> union(
-        ...     [
-        ...         filewise_index(['f1', 'f2']),
-        ...         segmented_index(['f1', 'f2'], [0, 0], [1, 1]),
-        ...     ]
-        ... )
-        MultiIndex([('f1', '0 days',               NaT),
-                    ('f2', '0 days',               NaT),
-                    ('f1', '0 days', '0 days 00:00:01'),
-                    ('f2', '0 days', '0 days 00:00:01')],
-                   names=['file', 'start', 'end'])
-
-    """
-    if not objs:
-        return filewise_index()
-
-    types = [index_type(obj) for obj in objs]
-
-    if len(set(types)) != 1:
-        objs = [to_segmented_index(obj) for obj in objs]
-
-    return union_misc(objs)
-
-
-def union_misc(
-    objs: typing.Sequence[pd.Index],
-) -> pd.Index:
-    r"""Create union of index objects.
-
-    Requires that levels and dtypes
+    If all index objects are conform to
+    :ref:`table specifications <data-tables:Tables>`
+    and at least one object is segmented,
+    the output is a segmented index.
```

Comment:
What happens if we have a mixture of filewise, segmented and misc indices?
I guess a misc index that is alike a segmented index, is also a segmented index?

Reply:
Exactly, and if is neither a filewise, nor a segmented index an error is raised because indices are not alike.